src/Beman/Optional26/tests/optional_constexpr.t.cpp

```diff
@@ -0,0 +1,1744 @@
+// src/Beman/Optional26/tests/optional_constexpr.t.cpp                -*-C++-*-
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <Beman/Optional26/optional.hpp>
+
+#include <functional>
+#include <ranges>
+#include <tuple>
+
+#include <gtest/gtest.h>
+
+TEST(OptionalConstexprTest, TestGTest) { ASSERT_EQ(1, 1); }
+
+namespace {
+struct empty {};
+struct no_default {
+    no_default()                             = delete;
+    no_default(const no_default&)            = default;
+    no_default(no_default&&)                 = default;
+    no_default& operator=(const no_default&) = default;
+    no_default& operator=(no_default&&)      = default;
+    no_default(empty) {};
+};
+
+struct base {
+    int i_;
+    constexpr base() : i_(0) {}
+    constexpr base(int i) : i_(i) {}
+};
+
+struct derived : public base {
+    int j_;
+    constexpr derived() : base(0), j_(0) {}
+    constexpr derived(int i, int j) : base(i), j_(j) {}
+};
+} // namespace
+
+TEST(OptionalConstexprTest, Constructors) {
+    constexpr beman::optional26::optional<int> i1;
+    constexpr beman::optional26::optional<int> i2{beman::optional26::nullopt};
+
+    constexpr int                              i  = 0;
+    constexpr beman::optional26::optional<int> i3 = i;
+    (void)i3;
+
+    constexpr beman::optional26::optional<empty> e1;
+    constexpr beman::optional26::optional<int>   e2{beman::optional26::nullopt};
```

Comment:
```suggestion
    constexpr beman::optional26::optional<empty>   e2{beman::optional26::nullopt};
```
?

Reply:
Probably want all of them. Not that the test is really interesting, but empty type constructed into an empty optional was intentional. I think.